docs/DeveloperGuide.adoc

```diff
@@ -295,131 +290,171 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * ` |busy student| refer and use the predictive text |input a command with great ease and less time
 
-|`* * *` |user |add a new person |
+|`* * ` |forgetful user |receive reminders on assignments and work near the deadline |remember to complete and submit it on time
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* *` |curious student |calculate and predict how much effort I need to put for a particular module based on the score I have gained so far |
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * ` |lazy and forgetful student |keep track of lectures that I have missed |
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * ` |student with too many assignments | automatically get my assignment sorted according to its grade percentage |
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* * ` |kiasu student |make a checklist of materials that I have to learn for my exams |
+
+|`*` |student interested in numbers |view an overall summary of my academic journey so far |reflect on it
+
+|`* ` |clumsy typer |have a program that still understand my commands even with minor spelling errors |not have to enter the command again
+
+|`* ` |calendar app user |sync my timetable to my own personal calendar |refer to my schedule
 |=======================================================================
 
-_{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+//(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Add module
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User request to add module
+2.  Modulo requests user to input the module's time schedule
+3.  User inputs the time schedule
+4.  Module adds the new module to the user's timetable
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+
+* 2a. The given time is invalid.
++
+[none]
+** 2a1. Modulo shows an error message.
++
+
+
+[discrete]
+=== Use case: Edit lecture time
+
+*MSS*
+
+1.  User request to edit the lecture time of a certain module
+2.  Modulo asks the user to input the new timing
+3.  User inputs the time
+4.  Module edits the lecture time in the user's timetable
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
+
+[none]
+
+* 2a. The given time is invalid.
++
+[none]
+** 2a1. Modulo shows an error message.
++
+* 4a. The module doesn't exist
 +
```

Comment:
Perhaps there should be an extension that handles the case where the input time clashes with the time of another existing lecture?